client_app/src/Shelters.js

```diff
@@ -0,0 +1,114 @@
+import { useState, useEffect } from "react";
+
+import ShelterList from "./Components/ShelterList";
+import { SERVER } from "./config";
+
+function Shelters() {
+  const [data, setData] = useState([]);
+  const [latitude, setLatitude] = useState(41.8781);
+  const [longitude, setLongitude] = useState(-87.6298);
+  const [radius, setRadius] = useState("10");
+  const [loading, setLoading] = useState(true);
+  const [isButtonDisabled, setButtonDisabled] = useState(true);
+  const [selectedShifts, setSelectedShifts] = useState([]);
+
+  let shelters_endpoint =
+    "https://api2-qa.gethelp.com/v2/facilities?page=0&pageSize=1000";
+  let volunteers_endpoint = "https://api2-qa.gethelp.com/volunteers/";
+
+  useEffect(() => {
+    setLoading(true);
+    let new_endpoint =
+      shelters_endpoint +
+      "&latitude=" +
+      latitude.toString() +
+      "&longitude=" +
+      longitude.toString() +
+      "&radius=" +
+      radius.toString();
+    fetch(new_endpoint, {
+      methods: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then(async (response) => (await response.json())["content"])
+      .then((shelters) => setData(shelters))
+      .catch((error) => console.log(error));
+  }, [latitude, longitude, radius, shelters_endpoint, volunteers_endpoint]);
+
+  function getLocation() {
+    setLoading(true);
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(setLocation);
+    }
+  }
+
+  function setLocation(location) {
+    setLatitude(location.coords.latitude);
+    setLongitude(location.coords.longitude);
+  }
+
+  function setRadiusfromLocation(event) {
+    setRadius(event.target.value);
+  }
+
+  function manageShifts(shifts) {
+    setSelectedShifts(shifts);
+    setButtonDisabled(selectedShifts.length === 0);
+  }
+
+  function submitShifts() {
+    let shifts = selectedShifts;
+    for (let i = 0; i < shifts.length; i++) {
+      shifts[i].worker = "submitted-volunteer@slu.edu";
+    }
```

Comment:
We will know who the volunteer is because the volunteer will be signed in and will have a "bearer token". This "bearer token" is usually passed to the server in the header of an HTTPS packet. The server will then contact GetHelp's authentication server to figure out the email address associated with this bearer token. Since the authentication is not yet implemented, you should place (hard code) the volunteer email address into the packet header as the 'Authorization' field (you have headers defined below on line 70). Then, on the server side you can add a function 'getUserFromToken', which for now will just read the 'Authorization' field and return the email address. Later, when authentication is implemented, we will change getUserFromToken function to contact GetHelp's authentication server.